### PR TITLE
docs: Move Core API training guide into a single file for consistency

### PR DESCRIPTION
**Pull request by tara-det-ai**
## Description

- [x] Changes have been manually QA'd
- [x] Help troubleshoot rstfmt linter issue.

### CODE REVIEW

File: docs/training/apis-howto/api-core-ug.rst [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+ Core API
+#########################
+
+In this guide, you'll learn how to use the Core API.
+
++------------------------------------------------------------------+
+| Visit the API reference                                          |
++==================================================================+
+| :doc:`/reference/reference-training/training/api-core-reference` |
++------------------------------------------------------------------+
```
Comment (rb-determined-ai) [inline docs/training/apis-howto/api-core-ug.rst:11]: does it make sense to put the API reference link at the end of the overview?

File: docs/training/apis-howto/overview.rst [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
+The Training API Guides describe how to use the Determined training APIs. 
+Each API guide contains a link to its corresponding API reference.
 
-Determined leverages specific APIs for each Deep Learning framework. In general, users convert their
+Determined leverages specific APIs for each Deep Learning framework. In general, you convert your
 existing training code by subclassing a Trial class and implementing methods that advertise
-components of the user's model - e.g., model architecture, data loader, optimizer, learning rate
-scheduler, callbacks, etc. This is called the Trial definition and by structuring their code in this
-way, Determined is able to run the training loop and provide advanced training and model management
+components of your model - e.g., model architecture, data loader, optimizer, learning rate
```
Comment (rb-determined-ai) [inline docs/training/apis-howto/overview.rst:10]: reword:

> subclassing a Trial class and implementing methods that advertise components of your model - e.g., model architecture, data loader, optimizer, learning rate scheduler, callbacks, etc.

To

> subclassing a Trial class and implementing methods that define to each component of training - e.g., model architecture, data loader, optimizer, learning rate scheduler, callbacks, etc.

Reason: The word "advertising" suggests a specific kind of software architecture that is completely unrelated to what is going on in a Trial class.

File: docs/training/apis-howto/overview.rst [hunk 3]
```diff
@@ -1,32 +1,33 @@
...
+#####################
+ Training API Guides
+#####################
 
-This section describes how to use the training APIs and contains the API reference information.
+The Training API Guides describe how to use the Determined training APIs. 
+Each API guide contains a link to its corresponding API reference.
 
-Determined leverages specific APIs for each Deep Learning framework. In general, users convert their
+Determined leverages specific APIs for each Deep Learning framework. In general, you convert your
```
Comment (rb-determined-ai) [inline docs/training/apis-howto/overview.rst:8]: This is out-of-date:

> Determined leverages specific APIs for each Deep Learning framework. In general, you convert your existing training code by subclassing a Trial class

Reason: it's describing the Trial APIs specifically, but this statement ignores the existence of the Core API.  Also, the usefulness of presenting the information this way may be limited to the near future.

Comment (rb-determined-ai) [reply docs/training/apis-howto/overview.rst:8]: ### Current state
- Core API
- High-level APIs
    - Trial APIs
        - PyTorchTrial
        - DeepSpeedTrial
        - *TFKerasTrial
        - *EstimatorTrial

(* = "destined for destruction")

### Future state:
- Core API
- High-level APIs
    - **Keras integration
    - **PytorchLightning integration
    - **Mosaic integration
    - Trial APIs
        - PyTorchTrial
        - DeepSpeedTrial (maybe?  not sure if we'll keep this one)

(** = "things Ryan expects we have in the future)

File: docs/training/apis-howto/overview.rst [hunk 4]
```diff
@@ -1,32 +1,33 @@
...
-way, Determined is able to run the training loop and provide advanced training and model management
+components of your model - e.g., model architecture, data loader, optimizer, learning rate
+scheduler, callbacks, etc. This is called the Trial definition. With the code structured in this way,
+Determined is able to run the training loop and provide advanced training and model management
 capabilities.
 
-Once users' model code are ported to Determined's APIs, they can use an :doc:`experiment
-configuration </reference/reference-training/experiment-config-reference>` to configure how
-Determined should train the model - e.g., multi-GPU, hyperparameter search, etc.
+Once your model code is ported to a Determined API, you can use an :doc:`experiment
```
Comment (rb-determined-ai) [inline docs/training/apis-howto/overview.rst]: The word "port" also suggests a Trial API, because it implies there's quite a bit of work in going from your current model to a model that works with Determined.  That's not exactly true of Core API, and we'd like it to also not be required with the eventual high-level APIs we are planning on writing.

File: docs/training/apis-howto/api-core-ug.rst [hunk 5]
```diff
@@ -0,0 +1,416 @@
...
+
+   .. code:: bash
+
+      det e create 4_distributed.yaml . -f
+
+The complete ``4_distributed.py`` and ``3_hpsearch.yaml`` listings used in this example can be found
+in the :download:`core_api.tgz </examples/core_api.tgz>` download or in the `Github repository
+<https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api>`_.
+
+
```
Comment (rb-determined-ai) [inline docs/training/apis-howto/api-core-ug.rst]: can you delete the extra empty lines at the bottom of this file?

File: docs/training/apis-howto/overview.rst (hunk 3 above)
Comment (tara-det-ai) [reply docs/training/apis-howto/overview.rst:8]: created [Docs Jira issue](https://determinedai.atlassian.net/browse/DOCS-1?atlOrigin=eyJpIjoiZWQ4Y2NlNjk5MWY1NDJlY2EyNWIwMTVhYzIwYTc5MDAiLCJwIjoiaiJ9) to fix this

File: docs/training/apis-howto/api-core-ug.rst (hunk 1 above)
Comment (tara-det-ai) [reply docs/training/apis-howto/api-core-ug.rst:11]: it does make sense. i'll keep them at the top for now because they act like a toggle, toggling back and forth between the guide and the corresponding ref doc. i'll add a task for myself to revisit all of our "ascii" tables because I need to reformat them. resolving comment.

https://determinedai.atlassian.net/browse/DOCS-9?atlOrigin=eyJpIjoiZmY4YzU1ZmVmOGZmNDBlNWI5YTQ1YjVmNTY4YTJmZGUiLCJwIjoiaiJ9

File: docs/training/apis-howto/overview.rst (hunk 2 above)
Comment (tara-det-ai) [reply docs/training/apis-howto/overview.rst:10]: rewritten

File: docs/training/apis-howto/overview.rst (hunk 4 above)
Comment (tara-det-ai) [reply docs/training/apis-howto/overview.rst]: i suggest changing **ported** to **converted**

e.g.,

Once your model code is converted, you can use ...

File: docs/training/apis-howto/api-core-ug.rst (hunk 5 above)
Comment (tara-det-ai) [reply docs/training/apis-howto/api-core-ug.rst]: seems to be resolved

File: docs/training/apis-howto/overview.rst [hunk 6]
```diff
@@ -1,33 +1,36 @@
...
+#####################
+
+Determined leverages specific training APIs for each Deep Learning framework. The Training API
+guides describe how to use these training APIs. Each API guide contains a link to its corresponding
+API reference.
+
+With the Core API, you can plug in your existing training code. With the Trial APIs, you first
+convert your existing training code by subclassing a Trial class and implementing methods that
+define each component of training - e.g., model architecture, data loader, optimizer, learning rate
+scheduler, callbacks, etc. This is called the Trial definition. With the code structured in this
```
Comment (rb-determined-ai) [inline docs/training/apis-howto/overview.rst:12]: > The Training API guides describe how to use these training APIs. Each API guide contains a link to its corresponding API reference.

Ooh, I like how clear this part is!

> Determined leverages specific training APIs for each Deep Learning framework ... With the Core API ... With the Trial APIs

This sentence structure suggests a taxonomy that doesn't reflect the actual taxonomy of our code.  The opening sentence, "Determined leverages specific training APIs for each Deep Learning framework" is really talking about Trial APIs.  Each trial is specific to a single deep learning framework.  In contrast, Core API is totally framework agnostic.  A sentence structure that might be more suitable might be like:

> Almost any deep learning model can be trained in Determined...  The Core API offers a flexible, low-level API for training models based in any framework...  The Trial APIs offer higher-level integrations with several specific popular frameworks...

Comment (tara-det-ai) [reply docs/training/apis-howto/overview.rst]: resolved with commit

File: docs/conf.py [hunk 7]
```diff
@@ -302,12 +302,12 @@ def setup(app):
...
-    "training-apis/api-core/hpsearch": "../../training/apis-howto/api-core/hpsearch.html",
-    "training-apis/api-core/index": "../../training/apis-howto/api-core/overview.html",
-    "training-apis/api-core/metrics": "../../training/apis-howto/api-core/metrics.html",
+    "training-apis/api-core/checkpoints": "../../training/apis-howto/api-core-ug.html",
+    "training-apis/api-core/distributed": "../../training/apis-howto/api-core-ug.html",
+    "training-apis/api-core/getting-started": "../../training/apis-howto/api-core-ug.html",
+    "training-apis/api-core/hpsearch": "../../training/apis-howto/api-core-ug.html",
+    "training-apis/api-core/index": "../../training/apis-howto/api-core-ug.html",
+    "training-apis/api-core/metrics": "../../training/apis-howto/api-core-ug.html",
     "training-apis/api-core/reference": "../../reference/reference-training/training/api-core-reference.html",
```
Comment (rb-determined-ai) [inline docs/conf.py:311]: last thing, can we also add links from the documents you are deleting to the new consolidated document?

Comment (tara-det-ai) [reply docs/conf.py:311]: more info needed

Comment (tara-det-ai) [reply docs/conf.py:311]: thank you

**Review by rb-determined-ai — APPROVED** [review]
Wonderful!
